From: Samuel Brian <samuel.brian@digi.com>
Date: Tue, 23 Apr 2019 14:58:10 +1000
Subject: mmcli: respect timeout for bearer commands

Previously, e.g.

  mmcli --timeout 5 --bearer $BIDX --disconnect

would not time out at the requested 5 seconds, or at the mmcli default of 30
seconds. It would time out after 25 seconds, which is a default timeout in
dbus.
---
 cli/mmcli-bearer.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/cli/mmcli-bearer.c b/cli/mmcli-bearer.c
index d589e6ed..f8494626 100644
--- a/cli/mmcli-bearer.c
+++ b/cli/mmcli-bearer.c
@@ -367,6 +367,9 @@ get_bearer_ready (GObject      *source,
                                            &ctx->manager,
                                            &ctx->object);
 
+    if (ctx->bearer)
+        mmcli_force_operation_timeout (G_DBUS_PROXY (ctx->bearer));
+
     if (info_flag)
         g_assert_not_reached ();
 
@@ -422,6 +425,9 @@ mmcli_bearer_run_synchronous (GDBusConnection *connection)
                                          &ctx->manager,
                                          &ctx->object);
 
+    if (ctx->bearer)
+        mmcli_force_operation_timeout (G_DBUS_PROXY (ctx->bearer));
+
     /* Request to get info from bearer? */
     if (info_flag) {
         g_debug ("Printing bearer info...");
